roles/vault/tasks/check.yml

```diff
@@ -1,33 +1,41 @@
 ---
-- name: Get vault health
-  ansible.builtin.uri:
-    validate_certs: "{{ dsc.exposedCA.type == 'none' }}"
-    url: https://{{ vault_domain }}/v1/sys/health?sealedcode=200&uninitcode=200
-    status_code: [200, 503]
-  register: vault_health
-  retries: 12
-  delay: 5
-  until: vault_health.json is defined
+- name: Get vault status
+  kubernetes.core.k8s_exec:
+    container: vault
+    pod: "{{ vault_pod }}"
+    namespace: "{{ dsc.vault.namespace }}"
+    command: vault status -format=json
+  register: vault_status
+  ignore_errors: true
 
-- name: Set vault_status to unavailable
+- name: Set vault_initialized fact
   ansible.builtin.set_fact:
-    vault_status: unavailable
-  when: vault_health.status == 503
+    vault_initialized: "{{ vault_status.stdout
+      | from_json
+      | dict2items
+      | selectattr('key', 'eq', 'initialized')
+      | map(attribute='value') | first | default('false') }}"
 
-- name: If Vault json is defined
-  when: vault_health.json is defined
-  block:
-    - name: Set vault_status to sealed
-      ansible.builtin.set_fact:
-        vault_status: sealed
-      when: vault_health.json.sealed
+- name: Set vault_sealed fact
+  ansible.builtin.set_fact:
+    vault_sealed: "{{ vault_status.stdout
+      | from_json
+      | dict2items
+      | selectattr('key', 'eq', 'sealed')
+      | map(attribute='value') | first | default('true') }}"
+
+- name: Set vault_status to "not init"
+  when: not vault_initialized
+  ansible.builtin.set_fact:
+    vault_status: not init
```

Comment:
Pas de risque d'interprétation en booléen ici avec le `not` ? Des guillemets me rassureraient :)

Reply:
Jamais eu le soucis mais je peux rajouter ça (;

Reply:
Pas dans un when. D'autant que vault_initialized est justement un booléen.
Comportement déjà testé et fonctionnel. :-)